### PR TITLE
don't create issue links to self

### PR DESCRIPTION
**Pull request by mocobeta**
#92 

Suppress creating link that points to the same page.

### CODE REVIEW

File: migration/src/jira_util.py [hunk 1]
```diff
@@ -289,34 +289,34 @@ def repl_att(m: re.Match):
     return text
 
 
-def embed_gh_issue_link(text: str, issue_id_map: dict[str, str]) -> str:
+def embed_gh_issue_link(text: str, issue_id_map: dict[str, str], gh_number_self: int = None) -> str:
```
Comment (mikemccand) [inline migration/src/jira_util.py:292]: I think Python itself does not enforce these type hints?

And I think I may've messed some of them up with copy/paste in past PRs.

I will try `mypy` static analysis and see what it says ;)

File: migration/src/jira_util.py [hunk 2]
```diff
@@ -289,34 +289,34 @@ def repl_att(m: re.Match):
...
     """Embed GitHub issue number
     """
     def repl_simple(m: re.Match):
         res = m.group(0)
         gh_number = issue_id_map.get(m.group(2))
-        if gh_number:
+        if gh_number and gh_number != gh_number_self:
```
Comment (mikemccand) [inline migration/src/jira_util.py:298]: Hmm `issue_id_map` is a `str -> str` dict type, but `gh_number` is an `int`?  Aren't we then comparing `str` to `int`?

File: migration/src/jira_util.py (hunk 1 above)
Comment (mocobeta) [reply migration/src/jira_util.py:292]: The default is invalid...

File: migration/src/jira_util.py (hunk 2 above)
Comment (mocobeta) [reply migration/src/jira_util.py:298]: Both `gh_number` and `gh_number_self` is `int`.

File: migration/src/jira_util.py (hunk 1 above)
Comment (mocobeta) [reply migration/src/jira_util.py:292]: Actually, the default value is not needed. https://github.com/apache/lucene-jira-archive/commit/614a3cd62be8ecf6207e9fb682cd0b376db051fd

File: migration/src/jira_util.py (hunk 2 above)
Comment (mocobeta) [reply migration/src/jira_util.py:298]: I fixed type hints in https://github.com/apache/lucene-jira-archive/commit/614a3cd62be8ecf6207e9fb682cd0b376db051fd

File: migration/src/jira_util.py (hunk 1 above)
Comment (mocobeta) [reply migration/src/jira_util.py:292]: @mikemccand Thanks for noticing this. I delayed checking type hits with `mypy` - will install and do static analysis.

**Pull request merged**